SimMuon/GEMDigitizer/src/GEMSignalModel.cc

```diff
@@ -0,0 +1,162 @@
+    double elecEff = 0.;
+    double partMom = hit.pabs();
+    double checkMuonEff = CLHEP::RandFlat::shoot(engine, 0., 1.);
+    double checkElecEff = CLHEP::RandFlat::shoot(engine, 0., 1.);
+    if (std::abs(hit.particleType()) == muonPdgId && checkMuonEff < averageEfficiency_)
+      digiMuon = true;
+    if (std::abs(hit.particleType()) != muonPdgId)  //consider all non muon particles with gem efficiency to electrons
+    {
+      if (partMom <= elecMomCut1)
+        elecEff = elecEffLowCoeff * std::exp(elecEffLowParam0 * partMom * momConvFact);
+      if (partMom > elecMomCut1 && partMom < elecMomCut2)
+        elecEff = elecEffMidCoeff * log(elecEffMidParam1 * partMom * momConvFact - elecEffMidParam0) /
+                  (elecEffMidCoeff + log(elecEffMidParam1 * partMom * momConvFact - elecEffMidParam0));
+      if (partMom > elecMomCut2)
+        elecEff = 1.;
+      if (checkElecEff < elecEff)
+        digiElec = true;
+    }
+    if (!(digiMuon || digiElec))
+      continue;
+    const int bx(getSimHitBx(&hit, engine));
+    const std::vector<std::pair<int, int> >& cluster(simulateClustering(top, &hit, bx, engine));
+    for (const auto& digi : cluster) {
+      detectorHitMap_.emplace(digi, &hit);
+      strips_.emplace(digi);
+    }
+  }
+}
+
+int GEMSignalModel::getSimHitBx(const PSimHit* simhit, CLHEP::HepRandomEngine* engine) {
+  int bx = -999;
+  const LocalPoint simHitPos(simhit->localPosition());
+  const float tof(simhit->timeOfFlight());
+  // random Gaussian time correction due to electronics jitter
+  float randomJitterTime = CLHEP::RandGaussQ::shoot(engine, 0., timeJitter_);
+  const GEMDetId id(simhit->detUnitId());
+  const GEMEtaPartition* roll(geometry_->etaPartition(id));
+  if (!roll) {
+    throw cms::Exception("Geometry") << "GEMSignalModel::getSimHitBx() - GEM simhit id does not match any GEM roll id: "
+                                     << id << "\n";
+    return 999;
+  }
+  if (roll->id().region() == 0) {
+    throw cms::Exception("Geometry")
+        << "GEMSignalModel::getSimHitBx() - this GEM id is from barrel, which cannot happen: " << roll->id() << "\n";
+  }
+  const int nstrips = roll->nstrips();
+  float middleStrip = nstrips / 2.;
+  const LocalPoint& middleOfRoll = roll->centreOfStrip(middleStrip);
+  const GlobalPoint& globMiddleRol = roll->toGlobal(middleOfRoll);
+  double muRadius = sqrt(globMiddleRol.x() * globMiddleRol.x() + globMiddleRol.y() * globMiddleRol.y() +
+                         globMiddleRol.z() * globMiddleRol.z());
+  double timeCalibrationOffset_ = geant_units::operators::convertCmToMm(muRadius) / cspeed;  //[ns]
+
+  const TrapezoidalStripTopology* top(dynamic_cast<const TrapezoidalStripTopology*>(&(roll->topology())));
+  const float halfStripLength(0.5 * top->stripLength());
+  const float distanceFromEdge(halfStripLength - simHitPos.y());
+
+  // signal propagation speed in material in [cm/ns]
+  double signalPropagationSpeedTrue = signalPropagationSpeed_ * cspeed;
```

Comment:
The units look wrong here. `cspeed` is in mm/ns, whereas the comment says that cm/ns are intended.

Reply:
Ah, I forgot about changing the comment.
Should I make another PR for it?

Reply:
If you wouldn't mind, it would be good to have the correct comment. I will just merge it once it is available.